File: NasleGhalam.ViewModels/City/CityViewModel.cs
using NasleGhalam.ViewModels._Attributes;
using System.ComponentModel.DataAnnotations;

namespace NasleGhalam.ViewModels.City
{
    public class CityViewModel
    {
        public int Id { get; set; }


        [Display(Name = "نام")]
        [Required(ErrorMessageResourceType = typeof(ErrorResources), ErrorMessageResourceName = "Required")]
        [MaxLength(50, ErrorMessageResourceType = typeof(ErrorResources), ErrorMessageResourceName = "MaxLen")]
        public string Name { get; set; }

        [Display(Name = "استان")]
        [RequiredDdlValidator(invalidValue: "0", ErrorMessageResourceType = typeof(ErrorResources), ErrorMessageResourceName = "RequiredDll")]
        public int ProvinceId { get; set; }


        public string ProvinceName { get; set; }

    }
}

File: NasleGhalam.ClientApp/src/viewModels/cityViewModel.js
import {
  required,
  displayName,
  maxLength,
  requiredDdl
} from 'plugins/vuelidate';

export default {
  cityObj: {
    Name: {
      displayName: displayName('نام'),
      maxLength: maxLength(50),
      required
    },
    ProvinceId: {
      displayName: displayName('استان'),
      requiredDdl: requiredDdl(0)
    }
  }
};
